src/components/modebar/modebar.js

```diff
@@ -178,12 +187,19 @@ proto.createIcon = function(thisIcon, name) {
     icon.setAttribute('width', (thisIcon.width / iconHeight) + 'em');
     icon.setAttribute('viewBox', [0, 0, thisIcon.width, iconHeight].join(' '));
 
-    var transform = name === 'toggleSpikelines' ?
-        'matrix(1.5 0 0 -1.5 0 ' + thisIcon.ascent + ')' :
-        'matrix(1 0 0 -1 0 ' + thisIcon.ascent + ')';
-
     path.setAttribute('d', thisIcon.path);
-    path.setAttribute('transform', transform);
+
+    if(thisIcon.transform) {
+        path.setAttribute('transform', thisIcon.transform);
+    }
+    else if(thisIcon.ascent !== undefined) {
+        // Legacy icon transform calculation
+        var transform = name === 'toggleSpikelines' ?
+            'matrix(1.5 0 0 -1.5 0 ' + thisIcon.ascent + ')' :
+            'matrix(1 0 0 -1 0 ' + thisIcon.ascent + ')';
```

Comment:
Thanks for moving `pull_font_svg` to the new `transform` property - it's definitely cleaner to calculate that upfront - and for maintaining the legacy version to support any users who *did* go to the trouble of figuring out `ascent` and `descent`. But I don't think we need to carry the `toggleSpikelines` hack over here - lets take it out, then we don't even need `name` in `createIcon` 🎉